components/ai_chat/resources/page/api/index.ts

```diff
@@ -18,6 +20,8 @@ export type State = Mojom.ServiceState & {
   isPremiumUserDisconnected: boolean
   isMobile: boolean
   isHistoryFeatureEnabled: boolean
+  isAgentProfileFeatureEnabled: boolean
```

Comment:
AIChatAgent

Reply:
That's really problematic imo because it means that nothing on the frontend matches up with the cpp code. I'm ok for this PR but only if we follow-up to change them all

Reply:
seems like frontend is not consistent, some `isContentAgent..`, `openAIChatAgenticProfile`, etc... 
`ContentAgent` should definitely be fixed in this PR, but follow-up for the rest is ok

Reply:
I'm fine with prefixing feature flags to AIChat, and "agentic"->"agent". 
But the use of "content agent" is because from the UIs point of view this profile is for content agent purposes. There will be other toggles for agentic tools that are available in all profiles, but this toggle is for content agent tools, which is only available in the special isolation profile, for now. The idea of what will be separated is evolving though...